### ISSUE TITLE
non-admin api users can update order after it is complete

### ISSUE THREAD

**Issue by carchrae** [solidusio/solidus#546]
we've noticed that a non-admin user can update an order through the api, even if it is in the complete state.  eg, using the order_token and making line_item adjustments.

i can't see why this should be allowed.  it means a user can change line items, etc.  (due to a bug in our UI code, this happened)

i locked down the order by adding the following can can rule:

`app/models/spree/ability_decorator.rb`

``` ruby
# See http://github.com/ryanb/cancan for more details on cancan.
require 'cancan'
class AbilityDecorator
  include CanCan::Ability

  def initialize(user)
    user ||= Spree.user_class.new

    if user.respond_to?(:has_spree_role?) && user.has_spree_role?('admin')
      # if you are an admin, we don't apply the rule in the else block
    else
      # not allowed to update completed orders - wtf, why isn't this a default!
      cannot [:update], Spree::Order do |order|
        order.completed?
      end
    end

  end
end

Spree::Ability.register_ability(AbilityDecorator)
```

now, clearly the UI code should note that the user's order is complete and should not allow them to update the order (yes, we had that bug, yes, we fixed the UI), but client side code should not check rules like this, so we added it to the server as well.

i'll file a pull request to update the can can rules, but i have to ask, was this a design choice to allow a user to change a completed order?  or is this simply a case that has been handled by controller logic before?

**Comment by seand7565** [comment]
Looks like this was fixed with #2787 - can anyone verify?

**Comment by kennyadsl** [comment]
@seand7565 yep, thanks for pointing out!
